src/lib.rs

```diff
@@ -105,105 +91,23 @@ pub enum PixelOrder {
     BGR, // Default
 }
 
-impl<SPI> Apa102<SPI>
-where
-    SPI: SpiBus,
-{
-    /// new constructs a controller for a series of APA102 LEDs.
-    /// By default, an End Frame consisting of 32 bits of zeroes is emitted
-    /// following the LED data. Control over the size and polarity
-    /// of the End Frame is possible using new_with_options().
-    /// PixelOrder defaults to BGR ordering, and can also be customized
-    /// using new_with_options()
-    pub fn new(spi: SPI) -> Apa102<SPI> {
-        Self {
-            spi,
-            end_frame_length: 4,
-            invert_end_frame: true,
-            pixel_order: PixelOrder::BGR,
-        }
-    }
-
-    pub fn new_with_options(
-        spi: SPI,
-        end_frame_length: u8,
-        invert_end_frame: bool,
-        pixel_order: PixelOrder,
-    ) -> Apa102<SPI> {
-        Self {
-            spi,
-            end_frame_length,
-            invert_end_frame,
-            pixel_order,
-        }
-    }
-
-    /// Free the owned resources consuming self
-    pub fn free(self) -> SPI {
-        self.spi
-    }
+#[path = "."]
+mod asynchronous {
```

Comment:
why was this renamed from asynch to asynchronous? Not saying it's necessarily wrong just wondering, has it been released with asynch previously? Perhaps an unnecessary inconvenience for consumers?

Reply:
It doesn't matter; this module isn't public (also there hasn't been a release with async yet). This PR makes no changes to the public API. The only symbol exported from this module is renamed in the line just below the module: `pub use asynchronous::Apa102 as Apa102Async`.

Reply:
To answer your question though: the [bisync documentation](https://docs.rs/bisync) has an `asynchronous` module in its example and I didn't bother changing that from the example.